x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/search_index/pipelines/ml_inference/target_field_help_text.tsx

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+
+import { EuiText } from '@elastic/eui';
+
+import { i18n } from '@kbn/i18n';
+
+import {
+  getMlModelTypesForModelConfig,
+  SUPPORTED_PYTORCH_TASKS,
+} from '../../../../../../../common/ml_inference_pipeline';
+import { TrainedModel } from '../../../../api/ml_models/ml_trained_models_logic';
+import { getMLType } from '../../../shared/ml_inference/utils';
+
+export interface TargetFieldHelpTextProps {
+  model?: TrainedModel;
+  pipelineName: string;
+  targetField: string;
+}
+
+export const TargetFieldHelpText: React.FC<TargetFieldHelpTextProps> = ({
+  pipelineName,
+  targetField,
+  model,
+}) => {
+  const baseText =
+    targetField.length > 0
+      ? i18n.translate(
+          'xpack.enterpriseSearch.content.indices.pipelines.addInferencePipelineModal.steps.configure.targetField.helpText.userProvided',
+          {
+            defaultMessage:
+              'This names the field that holds the inference result. It will be prefixed with "ml.inference", ml.inference.{targetField}',
+            values: {
+              targetField,
+            },
+          }
+        )
+      : i18n.translate(
+          'xpack.enterpriseSearch.content.indices.pipelines.addInferencePipelineModal.steps.configure.targetField.helpText.default',
+          {
+            defaultMessage:
+              'This names the field that holds the inference result. It will be prefixed with "ml.inference", if not set it will be defaulted to "ml.inference.{pipelineName}"',
+            values: {
+              pipelineName: pipelineName.length > 0 ? pipelineName : '<Pipeline Name>',
+            },
+          }
+        );
+  const modelType = model ? getMLType(getMlModelTypesForModelConfig(model)) : '';
+  if (modelType === SUPPORTED_PYTORCH_TASKS.TEXT_CLASSIFICATION) {
+    return (
+      <EuiText size="xs">
+        <p>{baseText}</p>
+        <p>
+          {i18n.translate(
+            'xpack.enterpriseSearch.content.indices.pipelines.addInferencePipelineModal.steps.configure.targetField.helpText.textClassificationModel',
+            {
+              defaultMessage:
+                'Additionally the predicted_value will be copied to "{fieldName}", if the prediction_probability is greater than 0.5.',
```

Comment:
This is not a change meant for this PR, but I wonder if we should make that `0.5` configurable 🤔 

This is a lot of text to put under one input box... Maybe making the 0.5 configurable would actually help with that.

Reply:
Also a minor sidenote: 0.5 should be localized as a FormattedNumber :)

Reply:
@brianmcgue I think we should make 0.5, and that its a lot of help text for a single input. Hopefully we can address both in 8.7 when we move this modal to a flyout or full page.